hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java

```diff
@@ -660,14 +676,14 @@ public DatanodeInfo getDatanodeInfo() {
     }
 
     synchronized ExecutorService initMoveExecutor(int poolSize) {
-      return moveExecutor = Executors.newFixedThreadPool(poolSize);
+      return moveExecutor = Executors.newFixedThreadPool(poolSize, new ThreadFactoryBuilder().setNameFormat("move-thread-").build());
     }
 
     synchronized ExecutorService getMoveExecutor() {
       return moveExecutor;
     }
 
-    synchronized void shutdownMoveExecutor() {
+    public synchronized void shutdownMoveExecutor() {
```

Comment:
can you include the comments from cdh5 about why we need these various changes? I just think it will be helpful if someone has to read this file in the future after we've archived cdh5 and needs more context about the reasons.

Reply:
Will do!